webClient/src/app/shared/editor-control/editor-control.service.ts

```diff
@@ -227,14 +228,40 @@ export class EditorControlService implements ZLUX.IEditor, ZLUX.IEditorMultiBuff
   }
 
   public closeFileHandler(fileContext: ProjectContext) {
+    console.log(`closeFileHandler start\n`);
     let cacheFileName = `${fileContext.model.fileName}:${fileContext.model.path}`;
     if (cacheFileName) {
       this.log.debug(`Clearing cache for`,cacheFileName);
       delete stateCache[cacheFileName];
     }
-    !fileContext.opened ? this.log.warn(`File ${fileContext.name} already closed.`) : fileContext.opened = false;
-    !fileContext.active ? this.log.warn(`File ${fileContext.name} already inactive.`) : fileContext.active = false;
+
+    /* dequeue dataset */
+    console.log(`closeFileHandler model File name is ${fileContext.model.fileName}`);
+    
+          /* Send DEQ dataset  */
+          let filePath = ['/', '\\'].indexOf(fileContext.model.path.substring(0, 1)) > -1 ? fileContext.model.path.substring(1) : fileContext.model.path;
+          const enqRequestUrl = ZoweZLUX.uriBroker.datasetEnqueueUri(filePath);   /* the ENQ URL */
+          let _observable = this.http.delete(enqRequestUrl);                        /* prepare to delete the ENQ */
+
+          console.log(`closeFileHandler delete request is for ${fileContext.model.fileName}`);
+
+          /* subscribe to the DELETE request */  
+          _observable.subscribe({
+            next: (response: any) => {
+              console.log(`closeFileHandler delete request OK`);              
+            },
+            error: (err) => {
+              // console.log(`closeFileHandler delete request FAILED, error status ${err.status}`); 
+              console.log(`closeFileHandler delete request FAILED.  Error is: `, err);             
+            }
+          });
+          /* end of subscribe */
+          
+    /* end of dequeue dataset */
+    !fileContext.opened ? this.log.warn(`File ${fileContext.model.fileName} already closed.`) : fileContext.opened = false;
+    !fileContext.active ? this.log.warn(`File ${fileContext.model.fileName} already inactive.`) : fileContext.active = false;
     fileContext.changed = false;
+    /* TBD is the line below correct? */
```

Comment:
The line below this one is used to get the value of the next file/tab up to display its value so I think it looks correct to me by getting a list of all open files, then filtering out from that list all the things that aren't the new file that is queued up to show. How it filters is by checking file name & file path combo which I think is probably a safe bet

Reply:
To test whether this works as intended, it could be useful to open a file, open more files, rename some actively opened files in the middle that aren't front-facing, close the files on top, and see if the UX for the files in the middle still works OK. But doesn't concern this PR too much anyway. Feel free to add this description to the comment if you'd like